### PR TITLE
Print source sentences if available

### PR DESCRIPTION
**Pull request by zdou0830**
#53

### CODE REVIEW

**Comment by zdou0830** [comment]
Thank you! I have fixed the merge errors. Please take another look!

File: compare_mt/reporters.py [hunk 1]
```diff
@@ -391,21 +392,39 @@ def print(self):
...
 
       print(f'--- {report_length} sentences where {sright}>{sleft} at {self.scorer.name()}')
       for bdiff, s1, s2, str1, str2, i in self.scorediff_lists[cnt][-report_length:]:
-          print (
-          f"{sleft}-{sright}={fmt(-bdiff)}, {sleft}={fmt(s1)}, {sright}={fmt(s2)}\n"
-          f"Ref:  {' '.join(ref[i])}\n"
-          f"{sleft}: {' '.join(out1[i])}\n"
-          f"{sright}: {' '.join(out2[i])}\n"
-        )
+        if self.src:
```
Comment (neubig) [inline compare_mt/reporters.py]: I think we should avoid having the duplicated code within the big if statement. We should try to have most of the code not be duplicated, and just the `Src:` part be within an `if`.

File: compare_mt/reporters.py [hunk 2]
```diff
@@ -417,22 +436,41 @@ def html_content(self, output_directory=None):
       ref, out1, out2 = self.ref, self.outs[left], self.outs[right]
       html = tag_str('h4', f'{report_length} sentences where {sleft}>{sright} at {self.scorer.name()}')
       for bdiff, s1, s2, str1, str2, i in self.scorediff_lists[cnt][:report_length]:
-        table = [
-          ['', 'Output', f'{self.scorer.idstr()}'],
-          ['Ref', ' '.join(ref[i]), ''],
-          [f'{sleft}', ' '.join(out1[i]), fmt(s1)],
-          [f'{sright}', ' '.join(out2[i]), fmt(s2)]
-        ]
+        if self.src:
```
Comment (neubig) [inline compare_mt/reporters.py]: Similarly here.

File: compare_mt/reporters.py [hunk 3]
```diff
@@ -417,22 +436,41 @@ def html_content(self, output_directory=None):
...
 
       html += tag_str('h4', f'{report_length} sentences where {sleft}>{sright} at {self.scorer.name()}')
       for bdiff, s1, s2, str1, str2, i in self.scorediff_lists[cnt][-report_length:]:
-        table = [
-          ['', 'Output', f'{self.scorer.idstr()}'],
-          ['Ref', ' '.join(ref[i]), ''],
-          [f'{sleft}', ' '.join(out1[i]), fmt(s1)],
-          [f'{sright}', ' '.join(out2[i]), fmt(s2)]
-        ]
+        if self.src:
```
Comment (neubig) [inline compare_mt/reporters.py]: Similarly here.

**Comment by zdou0830** [comment]
Thanks! I made changes accordingly. Please take another look.

**Comment by neubig** [comment]
Thanks!

**Pull request merged**